### PR TITLE
[TRRFFDA-150/TRRFFDA-151] Set patient to first stage at registration and move to the next when …

### PR DESCRIPTION
**Pull request by alintreznai**
…consent is signed

### CODE REVIEW

File: rdrf/rdrf/testing/unit/registration_tests.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            "state": "AU-QLD",
+            "postcode": "1234",
+            "phone_number": "5678",
+        }
+        patch_method = "rdrf.views.registration_rdrf.RdrfRegistrationView.is_recaptcha_valid"
+        with patch(patch_method, side_effect=lambda: True):
+            response = self.client.post(reverse("registration_register", kwargs={"registry_code": self.registry.code}), post_data)
+            self.assertEqual(response.status_code, 302)
+            return Patient.objects.filter(email=self.PATIENT_EMAIL).first()
+
```
Comment (sztamas) [inline rdrf/rdrf/testing/unit/registration_tests.py:82]: From here on the tests are not registration tests anymore as the class name suggests.

File: rdrf/rdrf/views/form_view.py [hunk 2]
```diff
@@ -1824,6 +1829,7 @@ def post(self, request, registry_code, patient_id, context_id=None):
             custom_consent_form.save()
             if patient_signature_form:
                 patient_signature_form.save()
+            self._move_to_next_stage(registry_model, patient_model)
```
Comment (sztamas) [inline rdrf/rdrf/views/form_view.py]: I don't think this is right.
1. The patient should be moved to `Eligibility` only if they have consented (ie. checked all the consent boxes)
2. Moving them to the next stage assumes that they are in the `Informed Consent` stage when the consent form is saved which isn't necessarily true.

I think we are looking towards having rules associated with stages that get evaluated when changes are made to a patient and decide whether the patient's stage should change.
In this case, the rules would be: current stage is 'Informed Consent' and all consent has been given.

At this stage of development, we could refrain going the full way, and just have a column with pre-defined rules on the `PatientStage`. ex. `PATIENT_REGISTERED`, `PATIENT_CONSENTED`.

Will see how things develop further down the line and whether we will have to give end-user more customisability by letting them provide their own rules.

**Review by sztamas — CHANGES_REQUESTED** [review]

File: rdrf/rdrf/testing/unit/registration_tests.py (hunk 1 above)
Comment (alintreznai) [reply rdrf/rdrf/testing/unit/registration_tests.py:82]: Updated naming.

File: rdrf/rdrf/views/form_view.py (hunk 2 above)
Comment (alintreznai) [reply rdrf/rdrf/views/form_view.py]: Updated.

**Review by sztamas — APPROVED** [review]

**Pull request merged**